drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableAttributeSelectorPopup.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.guided.dtable.client.widget.table;
+
+import org.drools.workbench.models.guided.dtable.shared.model.AttributeCol52;
+import org.drools.workbench.models.guided.dtable.shared.model.GuidedDecisionTable52;
+import org.drools.workbench.models.guided.dtable.shared.model.MetadataCol52;
+import org.drools.workbench.screens.guided.dtable.client.resources.i18n.GuidedDecisionTableConstants;
+import org.drools.workbench.screens.guided.rule.client.editor.AttributeSelectorPopup;
+import org.drools.workbench.screens.guided.rule.client.editor.RuleAttributeWidget;
+
+public class GuidedDecisionTableAttributeSelectorPopup extends AttributeSelectorPopup {
+
+    private final String[] existingAttributeNames;
+    private final GuidedDecisionTableView.Presenter presenter;
+
+    public GuidedDecisionTableAttributeSelectorPopup( String[] existingAttributeNames,
+                                                      GuidedDecisionTableView.Presenter presenter ) {
+        this.existingAttributeNames = existingAttributeNames;
+        this.presenter = presenter;
+
+        initialize();
+    }
+
+    @Override
+    protected String[] getAttributes() {
+        String[] attributes = RuleAttributeWidget.getAttributesList();
+        attributes[attributes.length] = GuidedDecisionTable52.NEGATE_RULE_ATTR;
```

Comment:
Yes it would have thrown an ```ArrayIndexOutOfBoundsException``` if the PR was merged "as is".. it didn't fail in the old code as we used a ```List``` not an array. It is probably not covered by your tests.